src/frontend/Sider/Behandling/Vilkårvurdering/utils.ts

```diff
@@ -1,119 +1,30 @@
-import { Vurdering } from '../vilkår';
-
-export function begrunnelseErPåkrevdOgUtfyllt(
-    svarsalternativ: Svaralternativ,
-    begrunnelse: Begrunnelse
-): boolean {
-    if (svarsalternativ.begrunnelseType === BegrunnelseRegel.PÅKREVD) {
-        return !begrunnelse || begrunnelse.trim().length > 0;
-    }
-    return true;
-}
-
-export const forventerBegrunnelsePåAlleSvarsalternativ = (regel: Regel) => {
-    const svaralternativ = Object.values(regel.svarMapping);
-    return (
-        svaralternativ.length > 0 &&
-        svaralternativ.every((svar) => svar.begrunnelseType !== BegrunnelseRegel.UTEN)
-    );
-};
-
-export function hentSvaralternativ(
-    regler: Regler,
-    vurdering: Vurdering
-): Svaralternativ | undefined {
-    if (!vurdering.svar) {
-        return undefined;
-    } else {
-        const regel = regler[vurdering.regelId];
-        return regel.svarMapping[vurdering.svar];
-    }
-}
-
-export function leggTilNesteIdHvis(
-    nesteRegelId: RegelId,
-    nySvarArray: Vurdering[],
-    hvisFunksjon: () => boolean
-): Vurdering[] {
-    const inneholderAlleredeNesteRegelId = nySvarArray.some((v) => v.regelId === nesteRegelId);
-    if (nesteRegelId !== 'SLUTT_NODE' && !inneholderAlleredeNesteRegelId && hvisFunksjon()) {
-        return [...nySvarArray, { regelId: nesteRegelId }];
+import { Delvilkår, RegelId, Delvilkårsett } from '../vilkår';
+
+export const finnAvhengighetTilOverordnetValg = (
+    vilkårsvurdering: Delvilkårsett,
+    regelenSomSkalSjekkes: RegelId
+): { følgerAvOverordnetValg: boolean; overordnetValgErOppfylt: boolean | undefined } => {
+    const følgerFraOverordnetValg = vilkårsvurdering[regelenSomSkalSjekkes].følgerFraOverordnetValg;
+
+    if (følgerFraOverordnetValg === null) {
+        return {
+            følgerAvOverordnetValg: false,
+            overordnetValgErOppfylt: undefined,
+        };
     }
-    return nySvarArray;
-}
-
-export const kanHaBegrunnelse = (svarsalternativ: Svaralternativ) =>
-    svarsalternativ.begrunnelseType !== BegrunnelseRegel.UTEN;
 
-export const oppdaterSvarIListe = (
-    nyttSvar: Vurdering,
-    vurderinger: Vurdering[],
-    beholdResterendeSvar = false,
-    beholdBeskrivelse = false
-): Vurdering[] => {
-    const { svar, regelId } = nyttSvar;
+    const { regel, svar } = følgerFraOverordnetValg;
 
-    const svarIndex = vurderinger.findIndex((s) => s.regelId === regelId);
-    const nySvarArray = vurderinger.slice(0, svarIndex);
-
-    if (!svar) {
-        return vurderinger;
-    }
-
-    const oppdatertNyttSvar = beholdBeskrivelse
-        ? { ...nyttSvar, begrunnelse: vurderinger[svarIndex].begrunnelse }
-        : nyttSvar;
-    return [...nySvarArray, oppdatertNyttSvar].concat(
-        beholdResterendeSvar ? [...vurderinger.slice(svarIndex + 1)] : []
-    );
+    return {
+        følgerAvOverordnetValg: true,
+        overordnetValgErOppfylt: vilkårsvurdering[regel].svar === svar,
```

Comment:
Ville endret navn på denne så det blir tydligere hva som sjekkes her.
```suggestion
        overordnetValgErBesvart: vilkårsvurdering[regel].svar === svar,
```